packages/dev/gui/src/3D/controls/touchHolographicV3Button.ts

```diff
@@ -0,0 +1,1028 @@
+                const textContainer = new Grid();
+                textContainer.addColumnDefinition(1);
+                textContainer.addRowDefinition(0.5);
+                textContainer.addRowDefinition(0.5);
+                textContainer.widthInPixels = totalPanelWidthInPixels;
+                textContainer.heightInPixels = 45;
+
+                const subtext = new TextBlock(`${this.name}_subtext`);
+                subtext.text = this._subtext;
+                subtext.color = "#EEEEEEAB";
+                subtext.fontSize = this.textSizeInPixels * 0.75;
+                subtext.fontWeight = "600";
+
+                if (this._imageUrl) {
+                    subtext.textHorizontalAlignment = Control.HORIZONTAL_ALIGNMENT_LEFT;
+                    subtext.paddingLeftInPixels = padding;
+                }
+
+                textContainer.addControl(text, 0);
+                textContainer.addControl(subtext, 1);
+                panel.addControl(textContainer);
+            } else {
+                panel.addControl(text);
+            }
+        }
+
+        contentContainer.addControl(panel);
+        return contentContainer;
+    }
+
+    // Mesh association
+    protected _createNode(scene: Scene): TransformNode {
+        this.name = this.name ?? "TouchHolographicV3Button";
+
+        const backPlateMesh = this._createBackPlate(scene);
+        const collisionMesh = this._createFrontPlate(scene);
+        const innerQuadMesh = this._createInnerQuad(scene);
+        const backGlowMesh = this._createBackGlow(scene);
+
+        this._frontPlateCollisionMesh = collisionMesh;
+        this._textPlate = <Mesh>super._createNode(scene);
+        this._textPlate.name = `${this.name}_textPlate`;
+        this._textPlate.isPickable = false;
+        this._textPlate.scaling.x = this.width;
+        this._textPlate.parent = collisionMesh;
+
+        this._backPlate = backPlateMesh;
+        this._backPlate.position = Vector3.Forward(scene.useRightHandedSystem).scale(this._backPlateDepth / 2);
+        this._backPlate.isPickable = false;
+
+        this._backPlate.addChild(collisionMesh);
+        this._backPlate.addChild(innerQuadMesh);
+
+        if (backGlowMesh) {
+            this._backPlate.addChild(backGlowMesh);
+        }
+
+        const tn = new TransformNode(`${this.name}_root`, scene);
+        this._backPlate.setParent(tn);
+
+        this.collisionMesh = collisionMesh;
+        this.collidableFrontDirection = this._backPlate.forward.negate(); // Mesh is facing the wrong way
+
+        return tn;
+    }
+
+    private _createBackPlate(scene: Scene) {
+        const backPlateMesh = CreateBox(`${this.name}_backPlate`, {}, scene);
+        backPlateMesh.isPickable = false;
+        backPlateMesh.visibility = 0;
+        backPlateMesh.scaling.z = 0.2;
+
+        SceneLoader.ImportMeshAsync(undefined, TouchHolographicV3Button.BACKPLATE_MODEL_BASE_URL, TouchHolographicV3Button.BACKPLATE_MODEL_FILENAME, scene).then((result) => {

```

Comment:
Does it make sense to track if the mesh initialization is done? i.e. when the button is actually ready to be rendered?

Reply:
Or maybe use async/await and only return when the mesh is ready? (if this part doesn't make sense i am fine with keeping the state as it is :-))

Reply:
Is this comment related to only the backplate mesh or all the meshes used to create the button?